access/grpc/grpc.go

```diff
@@ -1599,64 +1482,12 @@ func (c *BaseClient) SubscribeAccountStatusesFromLatestBlock(
 		return nil, nil, newRPCError(err)
 	}
 
-	accountStatutesChan := make(chan flow.AccountStatus)
-	errChan := make(chan error)
-
-	go func() {
-		defer close(accountStatutesChan)
-		defer close(errChan)
-		receiveAccountStatusesFromStream(ctx, subscribeClient, accountStatutesChan, errChan)
-	}()
-
-	return accountStatutesChan, errChan, nil
-}
-
-func receiveAccountStatusesFromStream[Stream interface {
-	Recv() (*executiondata.SubscribeAccountStatusesResponse, error)
-}](
-	ctx context.Context,
-	stream Stream,
-	accountStatutesChan chan<- flow.AccountStatus,
-	errChan chan<- error,
-) {
-	sendErr := func(err error) {
-		select {
-		case <-ctx.Done():
-		case errChan <- err:
-		}
+	convertAccountStatusResponse := func(response *executiondata.SubscribeAccountStatusesResponse) (flow.AccountStatus, error) {
+		return convert.MessageToAccountStatus(response)
 	}
+	accountStatutesChan, errChan := subscribe(ctx, subscribeClient.Recv, convertAccountStatusResponse, "account status")
 
-	var nextExpectedMsgIndex uint64
-	for {
-		accountStatusResponse, err := stream.Recv()
-		if err != nil {
-			if err == io.EOF {
-				// End of stream, return gracefully
-				return
-			}
-
-			sendErr(fmt.Errorf("error receiving account status: %w", err))
-			return
-		}
-
-		accountStatus, err := convert.MessageToAccountStatus(accountStatusResponse)
-		if err != nil {
-			sendErr(fmt.Errorf("error converting message to account status: %w", err))
-			return
-		}
-
-		if accountStatus.MessageIndex != nextExpectedMsgIndex {
```

Comment:
this check isn't included in `subscribe()`, but is still needed. you could probably generalize something that incorporates either messgeIndex or block height?

Reply:
This is a good catch. I came up with the second function for such cases. Please, take a look at it